File: module/common/src/main/java/com/github/jameshnsears/cameraoverlay/common/MethodLineLoggingTree.java
package com.github.jameshnsears.cameraoverlay.common;

import androidx.annotation.NonNull;

import timber.log.Timber.DebugTree;

public final class MethodLineLoggingTree extends DebugTree {
    @Override
    protected @NonNull
    String createStackElementTag(@NonNull final StackTraceElement element) {
        return String.format(
                "%s, %s",
                element.getMethodName(),
                element.getLineNumber());
    }
}

File: app/src/main/java/com/github/jameshnsears/cameraoverlay/common/MethodLineLoggingTree.kt
package com.github.jameshnsears.cameraoverlay.common

import timber.log.Timber

class MethodLineLoggingTree : Timber.DebugTree() {
    protected override fun createStackElementTag(element: StackTraceElement): String {
        return String.format(
            "CameraOverlay: %s.%s, %s",
            super.createStackElementTag(element),
            element.methodName,
            element.lineNumber
        )
    }
}
